### ISSUE TITLE
count_rewires in c_main.c

### ISSUE THREAD

**Issue by alan-stokes** [SpiNNakerManchester/sPyNNaker#575]
this is used in log_debugs, but looks like a provenance data item. It should be pushed into the provenance data stuff and thus make it more reportable in the main tools.

**Comment by pabogdan** [comment]
Very well spotted! Recording is something that I should have done a long time ago ...

And just to clarify do you mean recording (something that the user can request) or is this just something that would be added in the `reports` folder in some file somewhere?

**Comment by alan-stokes** [comment]
"And just to clarify do you mean recording" 
i didnt say recording, you did. 

provenance is anything that can be used to either report errors/warnings to the end user, or anything that can report how things behaved outside of the recorded variables of spikes, v, gsyn. Thus allowing a second run to deduce if it was functionally equivalent to the first. 

all provenance data items are put into a core level report inside reports/provenance_items. But anything that can be assessed to produce a error/ warning, aka dropped packets. are interpreted by the host during the provenance translation phase. Look at abstractPopulationMachineVertex for deeper understanding. 

basically stuff that's just being reported, doesnt have a report and message param in provenanceDataItem.
